File: KoohiiAite/src/main/java/com/diarmaidlindsay/koohii/model/Reading.java
package com.diarmaidlindsay.koohii.model;

/**
 * Represent an entry in the reading table
 */
public class Reading {
    private int id;
    private int heisigId;
    private String readingText;
    private int type; //0 = onyomi, 1 = kunyomi

    public Reading(int id, int heisigId, String readingText, int type) {
        this.id = id;
        this.heisigId = heisigId;
        this.readingText = readingText;
        this.type = type;
    }

    public int getId() {
        return id;
    }

    public int getHeisigId() {
        return heisigId;
    }

    public String getReadingText() {
        return readingText;
    }

    public int getType() {
        return type;
    }
}

File: app/src/main/java/tech/diarmaid/koohiiaite/database/entity/Reading.kt
package tech.diarmaid.koohiiaite.database.entity

import androidx.annotation.NonNull
import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

/**
 * Represent an entry in the reading table
 */
@Entity(tableName = "reading",
        foreignKeys = [ForeignKey(entity = HeisigKanji::class, parentColumns = ["id"], childColumns = ["heisig_id"])])
data class Reading(@PrimaryKey val id: Int,
                   @ColumnInfo(name = "heisig_id") val heisigId: Int,
                   @ColumnInfo(name = "reading_text", defaultValue = "") @NonNull val readingText: String?,
                   val type: Int = 0
)
